### PR TITLE
Documentations

### PR DESCRIPTION
**Pull request by Thanatoz-1**
[Draft]

This MR adds docs and their respective web objects in descending order.  
Docs completed so far are: 
- [x] Trim
- [x] Time stretch
- [ ] Time Mask
- [x] Tanh Distortion
- [ ] Shift
- [ ] Seven band parametric eq
- [ ] Room simulator
- [ ] Reverse
- [ ] Resample
- [ ] Polarity inversion
- [ ] Pitch Shift
- [ ] Peaking filter
- [ ] Padding
- [ ] Normalize
- [ ] mp3 compression
- [ ] Low shelf filter 
- [ ] Low pass filter 
- [ ] Loudness normalization
- [ ] Limiter
- [ ] Lambda
- [ ] High shelf filter
- [ ] High pass filter
- [ ] Gain
- [ ] Gain transition
- [ ] Clipping distortion
- [ ] Clip
- [ ] Band stop filter 
- [ ] Band pass filter
- [ ] Apply impulse response
- [ ] Air absoption
- [ ] Add short noises
- [x] Add gaussian SNR
- [x] Add gaussian noise
- [x] Add background noise

### CODE REVIEW

**Comment by Thanatoz-1** [comment]
Hey @iver56 ,
I am pushing my first documentation [Trim] here. I am considering a document to be complete only if it contains the waveform & spectrogram comparison, and audio samples. Please correct me wherever I might be wrong or else I will continue.

File: docs/waveform_transforms/trim.md [hunk 1]
```diff
@@ -2,4 +2,34 @@
...
+    top_db=30.0,
+    p=1.0
+)
+
+augmented_sound = transform(my_waveform_ndarray, sample_rate=16000)
+```
+
+## Trim API
+
+[`top_db`](#top_db){ #top_db }: `float`
```
Comment (iver56) [inline docs/waveform_transforms/trim.md:31]:
```suggestion
[`top_db`](#top_db){ #top_db }: `float` • unit: Decibel
```

File: docs/waveform_transforms/trim.md [hunk 2]
```diff
@@ -2,4 +2,34 @@
...
+
+## Input-output examples
+
+![Input-output waveforms and spectrograms](Trim.webp)
+
+| Input sound                                                                             | Transformed sound                                                                             |
+|-----------------------------------------------------------------------------------------|-----------------------------------------------------------------------------------------------|
+| <audio controls><source src="../AddGaussianNoise_input.flac" type="audio/flac"></audio> | <audio controls><source src="../Trim_transformed.flac" type="audio/flac"></audio> |
+
+## Usage examples
```
Comment (iver56) [inline docs/waveform_transforms/trim.md:16]:
```suggestion
## Usage example
```

**Review by iver56 — CHANGES_REQUESTED** [review]
Thanks :) Can you also include your changes in generate_examples_for_doc.py?

**Comment by Thanatoz-1** [comment]
Certainly, I'll also add changes in the generate_examples_for_doc.py

**Comment by iver56** [comment]
I just merged some of my own documentation updates btw: https://github.com/iver56/audiomentations/pull/237

**Comment by Thanatoz-1** [comment]
Thanks, Taking a pull now and updating the MR too.

**Comment by iver56** [comment]
I will wait for your changes in generate_examples_for_doc.py, and then I think this is ready to be merged

**Comment by iver56** [comment]
One thing I'm not sure if I like is the dependency on AddGaussianNoise_input.flac

There are pros and cons

Pro:

* We save space in the repository and maybe save some network bandwidth for uses that browse several documentation pages (because cache)

Con:

* Maintenance becomes harder. For example, if I update the audio example in the AddGaussianNoise documentation, other pages that rely on AddGaussianNoise_input.flac will be broken

What do you think? Does the pro outweight the con?

**Comment by Thanatoz-1** [comment]
Hey @iver56,

Sorry for not being able to reply sooner. I tried to think over this trade-off and for now, I feel let's try to avoid cons and stick to two separate files for each transform (input and output).  I've modified the `generate_examples_for_docs.py` and have added most of the transforms for now. I'll push all the changes in some time and then you can merge.

**Review by iver56 — APPROVED** [review]

**Comment by iver56** [comment]
Thanks 👍

**Comment by Thanatoz-1** [comment]
Also, for the remaining documents, I will be creating another MR and update the rest of it asap.

**Pull request merged**